`Dropdown::ListItem::Interactive` - Add support for trailing icon (#2042)
hashicorp/design-system: website/docs/components/dropdown/partials/guidelines/guidelines.md

Icons in ListItems are optional. Generally, we recommend letting the text speak for itself, but icons can add value in the following situations:

- When they reinforce the content, e.g., `edit` for an edit or rename action. 

> - When using Interactive ListItems: using trailingIcons to indicate a link is external.

Reviewer: Not overly opinionated on this, but I think this reads a bit better as it's more of a sentence.

```suggestion
- When using Interactive ListItems, a `trailingIcon` can indicate that a link is external.
```

  <Doc::ListContainer class="hds-dropdown__list">
    <Hds::Dropdown::ListItem::Interactive @text="Rename cluster" @color="action" @icon="edit" />
    <Hds::Dropdown::ListItem::Interactive @text="Restore cluster" @color="action" @icon="reload" />

>     <Hds::Dropdown::ListItem::Interactive @text="Reference cluster" @color="action" @icon="github" @trailingIcon="external-link" />

Reviewer 2: @didoo for some reason this isn't showing up in the example. Clearly I'm missing something.

Reviewer 3: it's there, [have a peek](https://hds-website-git-dropdown-listitem-interactive-72c92b-hashicorp.vercel.app/components/dropdown#icon-usage). if you tested this on your machine, remember that since v2 changes you'd need to run `yarn start` in `packages/components` alongside with running the `website` with `ember s` – this is to make sure the latest changes in the components library are compiled and ready to be used on the website.

Reviewer 2: I'm a goof ball lol It's ALWAYS this step that gets me lol Thanks for the reminder.
